consensus/safety-rules/src/safety_rules.rs

```diff
@@ -302,16 +302,20 @@ impl SafetyRules {
             .cloned()
             .ok_or(Error::InvalidLedgerInfo)?;
 
+        // Update the waypoint to a newer value, this might still be stale than the current epoch.
+        let waypoint = &Waypoint::new_epoch_boundary(ledger_info)
+            .map_err(|error| Error::InternalError(error.to_string()))?;
+        self.persistent_storage.set_waypoint(waypoint)?;
+
         let current_epoch = self.persistent_storage.safety_data()?.epoch;
+        if current_epoch > epoch_state.epoch {
+            return Err(Error::NotInitialized(format!(
```

Comment:
Probably should introduce a new error type, but we can make that a bootcamp task.